bumble/device.py

```diff
@@ -433,37 +440,344 @@ def is_directed(self):
+                ),
+                advertising_sid=advertising_parameters.advertising_sid,
+                scan_request_notification_enable=(
+                    1 if advertising_parameters.enable_scan_request_notifications else 0
+                ),
+            ),
+            check_result=True,
+        )
+        self.selected_tx_power = response.return_parameters.selected_tx_power
+        self.advertising_parameters = advertising_parameters
+
+    async def set_advertising_data(self, advertising_data: bytes) -> None:
+        # pylint: disable=line-too-long
+        await self.device.send_command(
+            HCI_LE_Set_Extended_Advertising_Data_Command(
+                advertising_handle=self.advertising_handle,
+                operation=HCI_LE_Set_Extended_Advertising_Data_Command.Operation.COMPLETE_DATA,
+                fragment_preference=HCI_LE_Set_Extended_Advertising_Parameters_Command.SHOULD_NOT_FRAGMENT,
+                advertising_data=advertising_data,
+            ),
+            check_result=True,
+        )
+        self.advertising_data = advertising_data
+
+    async def set_scan_response_data(self, scan_response_data: bytes) -> None:
+        # pylint: disable=line-too-long
+        if (
+            scan_response_data
+            and not self.advertising_parameters.advertising_event_properties.is_scannable
+        ):
+            logger.warning(
+                "ignoring attempt to set non-empty scan response data on non-scannable "
+                "advertising set"
+            )
+            return
+
+        await self.device.send_command(
+            HCI_LE_Set_Extended_Scan_Response_Data_Command(
+                advertising_handle=self.advertising_handle,
+                operation=HCI_LE_Set_Extended_Advertising_Data_Command.Operation.COMPLETE_DATA,
+                fragment_preference=HCI_LE_Set_Extended_Advertising_Parameters_Command.SHOULD_NOT_FRAGMENT,
+                scan_response_data=scan_response_data,
+            ),
+            check_result=True,
+        )
+        self.scan_response_data = scan_response_data
+
+    async def set_periodic_advertising_parameters(
+        self, advertising_parameters: PeriodicAdvertisingParameters
+    ) -> None:
+        # TODO: send command
+        self.periodic_advertising_parameters = advertising_parameters
+
+    async def set_periodic_advertising_data(self, advertising_data: bytes) -> None:
+        # TODO: send command
+        self.periodic_advertising_data = advertising_data
+
+    async def set_random_address(self, random_address: Address) -> None:
+        await self.device.send_command(
+            HCI_LE_Set_Advertising_Set_Random_Address_Command(
+                advertising_handle=self.advertising_handle,
+                random_address=(random_address or self.device.random_address),
+            ),
+            check_result=True,
+        )
+
+    async def start(
+        self, duration: float = 0.0, max_advertising_events: int = 0
```

Comment:
nit: Need some docstring to describe units and behaviors.

Reply:
done